### PR TITLE
[IPAM] add documentation on IPAM CRDs

### PR DESCRIPTION
**Pull request by stuggi**
Depends-On: https://github.com/openstack-k8s-operators/infra-operator/pull/40
Jira: OSP-19996

### CODE REVIEW

File: networking.md [hunk 1]
```diff
@@ -506,4 +506,166 @@ spec:
 Additionally there is a CRD-less `service_controller`. It watches for `LoadBalancer` services in the same namespace of a `DNSMasq` instance and creates a `DNSData` CR to auto register service endpoints which have an annotation set with  `"dnsmasq.network.openstack.org/hostname": <hostname>`. This annotation gets auto added to the `LoadBalancer` services exposed using lib-common `https://github.com/openstack-k8s-operators/lib-common/pull/258`.
 
 ## IPAM
-TBD
\ No newline at end of file
+The the infra-operator privides CRDs for IPAM.
+
+The `NetConfig` CRD allows do describe the overall networks used for the environment.
```
Comment (hjensas) [inline networking.md:511]: nit: typo s/do/to/

File: networking.md [hunk 2]
```diff
@@ -506,4 +506,166 @@ spec:
...
+
+* a `network` described by `name`, `mtu` (default 1500), and one or multiple subnets.
+* a `subnet` describedc by `name`, one or multiple `allocationRanges` with `start` and `end`, `cidr`, `vlan` id, `gateway` and `routes`. Using `excludeAddresses` individual IP addresses from within the `cidr` can be excluded to be assigned (dynamic or static).
```
Comment (hjensas) [inline networking.md:514]: nit: typo s/describedc/described/

Generally the sentence is a bit hard to parse with comma and comma and. A suggestion that is a bit more words but possibly more readable. Alternatively we could use a "described by:" pattern and then bullet list explaining each field?

* a `subnet` described by: `name`, `cidr`, `vlan` (id), `gateway`, `routes`, `excludeAddresses` and `allocationRanges`. One or multiple `allocationRanges` can be defined, each with a `start` and `end` address. `routes` are optional, but one or multiple can be defined. Each route defining a `destination` network and `nexthop` router address. Using `excludeAddresses` individual IP addresses from within the `cidr` can be excluded to be assigned (dynamic or static).

File: networking.md [hunk 3]
```diff
@@ -506,4 +506,166 @@ spec:
...
+    vlan: 30
+  - address: 172.20.0.10
+    cidr: 172.20.0.0/24
+    mtu: 1500
+    network: Tenant
+    subnet: subnet1
+    vlan: 50
+```
+
+The reservation is also storted via the `Reservation` CRD. The CRD is meant to be "internal" and not direct used.
```
Comment (hjensas) [inline networking.md:671]: nit: s/storted/stored/

I think using comma nd refering to `Reservation` CRD as "it" would be better
 s/... `Reservation` CRD. The CRD is meant to be "internal" and not direct used./... `Reservation` CRD, but it is intended for "internal" use only.

File: networking.md [hunk 4]
```diff
@@ -506,4 +506,166 @@ spec:
...
+    subnets:
+    - name: subnet1
+      allocationRanges:
+      - end: 172.20.0.250
+        start: 172.20.0.10
+      cidr: 172.20.0.0/24
+      vlan: 50
+```
+
+The `IPSet` CRD can be used by a requestor to register an IP address.
```
Comment (hjensas) [inline networking.md:583]: Is "an IP address" correct? - It is registering IP addresses on one or more networks?

File: networking.md [hunk 5]
```diff
@@ -506,4 +506,166 @@ spec:
...
+
+The request is a list of networks with `name` and `subnetName`, optional `defaultRoute` for one subnet in total. Using the `fixedIP` a specific IP can be requested, if not already assigned. The `fixedIP` don't have to be from within an `allocationRange`.
+
+The status if the `IPSet` will have all details about the reserved IPs and network details to be used by the requestor. The `IPSet` can be used by the request if the overall `Ready` condition is `True`.
```
Comment (hjensas) [inline networking.md:587]: s/by the request/by the requestor/ ?

File: networking.md (hunk 4 above)
Comment (stuggi) [reply networking.md:583]: right

**Review by hjensas — COMMENTED** [review]
LGTM +1

**Review by abays — APPROVED** [review]
/lgtm
